### PR TITLE
Various code refactor in project and tests

### PR DESCRIPTION
**Pull request by btry**
Contains code refactor done in my work for policies feedback. To distinguish tasks and make the other PR easier to review once ready

### CODE REVIEW

File: inc/agent.class.php [hunk 1]
```diff
@@ -42,7 +42,10 @@ class PluginFlyvemdmAgent extends CommonDBTM implements PluginFlyvemdmNotifiable
    const ENROLL_INVITATION_TOKEN = 1;
    const ENROLL_ENTITY_TOKEN     = 2;
 
-   const DEFAULT_TOKEN_LIFETIME  = "P7D";
+   const DEFAULT_TOKEN_LIFETIME  = 'P7D';
+
+   const MINIMUM_ANDROID_VERSION = '2.0';
+   const MINIMUM_APPLE_VERSION = '1.0';
```
Comment (btry) [inline inc/agent.class.php]: introduce constants. They are also used for the unit tests to reduce maitnenance.

File: inc/common.class.php [hunk 2]
```diff
@@ -145,4 +145,28 @@ public static function getMax(CommonDBTM $item, $condition, $fieldName) {
...
+    * @param string $haystack
+    * @param string $needle
+    * @return bool
+    */
+   public static function endsWith($haystack, $needle) {
+      // search forward starting from end minus needle length characters
+      return $needle === '' || (($temp = strlen($haystack) - strlen($needle)) >= 0 && strpos($haystack,
+         $needle, $temp) !== false);
+   }
+
```
Comment (btry) [inline inc/common.class.php]: moved from installer class. Used in the MQTT daemon

File: tests/suite-integration/Entity.php [hunk 3]
```diff
@@ -31,7 +31,7 @@
 
 namespace tests\units;
 
-use Glpi\Test\CommonTestCase;
+use Flyvemdm\Test\CommonTestCase;
```
Comment (btry) [inline tests/suite-integration/Entity.php]: Flyvemdm\Test\CommonTestCase inherits from Glpi\Test\CommonTestCase and will allow us to add some helper methods for unit tests

File: tests/suite-integration/PluginFlyvemdmAgent.php [hunk 4]
```diff
@@ -32,18 +32,13 @@
...
 
 /**
  * TODO: testDeviceCountLimit should go to Entity unit test, remove inline engine when change that.
  */
 class PluginFlyvemdmAgent extends CommonTestCase {
 
-   /**
-    * @var string
-    */
-   private $minAndroidVersion = '2.0.0';
```
Comment (btry) [inline tests/suite-integration/PluginFlyvemdmAgent.php]: replaced by the constant in the agent class

File: tests/suite-integration/PluginFlyvemdmAgent.php [hunk 5]
```diff
@@ -179,7 +173,7 @@ public function testEnrollAgent() {
       // Test enrollment with a too low version
       $rows = $invitationLog->find("1");
       $agent = $this->agentFromInvitation($user, $guestEmail, $serial, $invitationToken, 'android',
-         '1.9');
+         '1.9.0');
```
Comment (btry) [inline tests/suite-integration/PluginFlyvemdmAgent.php]: Must be semver (3 numbers separated by a dot)

File: tests/suite-integration/PluginFlyvemdmAgent.php [hunk 6]
```diff
@@ -863,54 +857,6 @@ private function wipeDevice(\PluginFlyvemdmAgent $agent, $wipe = true, $expected
...
-      $this->setupGLPIFramework();
-      unset($_REQUEST['user_token']);
-
-      // Try to enroll
-      $agent = $this->newTestedInstance();
-      $agent->add($input);
-
-      return $agent;
-   }
-
```
Comment (btry) [inline tests/suite-integration/PluginFlyvemdmAgent.php]: a lot of  test classes uses this code. Factorizing...

File: tests/suite-integration/PluginFlyvemdmAgent.php [hunk 7]
```diff
@@ -863,54 +857,6 @@ private function wipeDevice(\PluginFlyvemdmAgent $agent, $wipe = true, $expected
       $this->string($row['message'])->isEqualTo(json_encode($mqttMessage, JSON_UNESCAPED_SLASHES));
    }
 
-   /**
-    * Create a new invitation
-    *
-    * @param string $guestEmail
-    * @return \PluginFlyvemdmInvitation
-    */
-   private function createInvitation($guestEmail) {
```
Comment (DIOHz0r) [inline tests/suite-integration/PluginFlyvemdmAgent.php]: I don't see were do you move this method, this is necesary for the enrollment process

File: tests/suite-integration/PluginFlyvemdmAgent.php [hunk 8]
```diff
@@ -863,54 +857,6 @@ private function wipeDevice(\PluginFlyvemdmAgent $agent, $wipe = true, $expected
...
-    *
-    * Try to enroll an device by creating an agent. If the enrollment fails
-    * the agent returned will not contain an ID. To ensore the enrollment succeeded
-    * use isNewItem() method on the returned object.
-    *
-    * @param \User $user
-    * @param array $input enrollment data for agent creation
-    * @return object
-    */
-   private function enrollFromInvitation(\User $user, array $input) {
```
Comment (DIOHz0r) [inline tests/suite-integration/PluginFlyvemdmAgent.php]: same comment here

File: tests/suite-integration/PluginFlyvemdmAgent.php [hunk 9]
```diff
@@ -967,81 +913,13 @@ private function loginAndAddAgent(array $currentInvitation) {
...
-    * @param \User $user object
-    * @param string $guestEmail
-    * @param string|null $serial if null the value is not used
-    * @param string $invitationToken
-    * @param string $mdmType
-    * @param string|null $version if null the value is not used
-    * @param string $inventory xml
-    * @return object
-    */
-   private function agentFromInvitation(
```
Comment (DIOHz0r) [inline tests/suite-integration/PluginFlyvemdmAgent.php]: same comment here

File: tests/suite-integration/PluginFlyvemdmAgent.php [hunk 10]
```diff
@@ -967,81 +913,13 @@ private function loginAndAddAgent(array $currentInvitation) {
...
          'inventory'         => TestingCommonTools::AgentXmlInventory($serial),
       ]);
       return $agentId;
    }
 
-   /**
-    * @param string $userIdField
-    * @return array
-    */
-   private function createUserInvitation($userIdField) {
```
Comment (DIOHz0r) [inline tests/suite-integration/PluginFlyvemdmAgent.php]: same comment here

File: tests/suite-integration/PluginFlyvemdmAgent.php (hunk 7 above)
Comment (btry) [reply tests/suite-integration/PluginFlyvemdmAgent.php]: the methid moved in  the test pachage (see the refactor branch, planned to merge into master if these tests pass)

File: tests/suite-integration/PluginFlyvemdmAgent.php (hunk 8 above)
Comment (btry) [reply tests/suite-integration/PluginFlyvemdmAgent.php]: same for all other places...

File: tests/suite-integration/PluginFlyvemdmAgent.php (hunk 7 above)
Comment (btry) [reply tests/suite-integration/PluginFlyvemdmAgent.php]: here : https://github.com/flyve-mdm/backend-tests/blob/refactor/src/Flyvemdm/Test/CommonTestCase.php#L42

Comment (btry) [reply tests/suite-integration/PluginFlyvemdmAgent.php]: as discussed today, now all t e st helper mehods moved in  the folter tests/src.

Package backend-tests becomes obsolete

**Pull request merged**